server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CloudFormationNodeVisitor.java

```diff
@@ -45,85 +54,118 @@
 
     private final Logger logger;
     private CloudFormationModule cfnModule;
+    private Graph<RootNode, RootRelationship> topology;
 
     /**
      Creates a <tt>CloudFormationNodeVisitor<tt> in order to build a template with the given
      <tt>CloudFormationModule<tt>.
 
-     @param logger    Logger for logging visitor behaviour
+     @param context TransformationContext to extract topology and logger
      @param cfnModule Module to build the template model
      */
-    public CloudFormationNodeVisitor(Logger logger, CloudFormationModule cfnModule) {
-        this.logger = logger;
+    public CloudFormationNodeVisitor(TransformationContext context, CloudFormationModule cfnModule) {
+        this.logger = context.getLogger(getClass());
+        this.topology = context.getModel().getTopology();
         this.cfnModule = cfnModule;
     }
 
+    /**
+     Get the Compute node this webApplication is ultimately hosted on
+
+     @param webApplication the webApplication to find the host for
+     @return the underlying Compute node
+     */
+    protected static Compute getCompute(WebApplication webApplication) {
```

Comment:
I feel like `CloudFormationNodeVisitor` is getting pretty big. 
Maybe we should extract some of the utility methods like these `getCompute` methods? 
We might need these for other future visitors, so it might be a good idea to extract them regardless of the size of this class.

Reply:
I extracted these into a parent class